src/Microsoft.Health.Blob/Features/Storage/BlobClientFactory.cs

```diff
@@ -26,8 +26,8 @@ public static BlobServiceClient Create(BlobDataStoreConfiguration configuration)
 
             if (configuration.AuthenticationType == BlobDataStoreAuthenticationType.ManagedIdentity)
             {
-                var defaultCredentials = new DefaultAzureCredential();
-                return new BlobServiceClient(new Uri(configuration.ConnectionString), defaultCredentials, blobClientOptions);
+                var managedIdentityCredential = new ManagedIdentityCredential(clientId: configuration.UserAssignedManagedIdentityAppId);
```

Comment:
Just for my understanding, is there any possibility that configuration.UserAssignedManagedIdentityAppId is null and we want to use defaultCredentials for ManagedIdentity?

Reply:
I would say that is a valid use case. I would think this is more of an override.

Reply:
When it's null it should revert to using a system assigned managed identity. DefaultCredentials will try other things like VS credentials, environment credentials, etc. I can make it use DefaultCredentials if ManagedIdentity is null or empty, if that's what we want.

Reply:
I would say we should probably fallback to using `DefaultAzureCredential` if the `ManagedIdentityClientId` is null or empty. My main reason is that otherwise this is a breaking change. 

Would it be easier to simply specify the clientId in the options for `DefaultAzureCredential` like in this example? https://docs.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet#examples

Reply:
Makes sense. Yeah that looks easier, let me give it a try.